tests/BlockChypTest/BlockChypTest.csproj

```diff
@@ -7,6 +7,8 @@
   </PropertyGroup>
 
 <ItemGroup>
+  <PackageReference Include="FsCheck" Version="2.14.0" />

```

Comment:
I'm not recommending that you guys write all tests with `FsCheck`, but I personally think it's great. 

Well-written property-based tests tend to provide higher assurance and help find bugs in your implementation that traditional unit tests otherwise wouldn't. Unit tests, however, also have their place.

Reply:
Again, this is just a suggestion and I can remove it, but I think that you'll see the usefulness by taking a look at the tests I've added.